[Bug]: Firefox Nightly crashes when opening a new tab after accessing about:config

### Steps to reproduce

1. Open the Firefox Nightly browser
2. Type "about:config" in the address bar
3. Press "Enter"
4. Open a new tab
5. Observe the behaviour

### Expected behaviour

New tab is opened.

### Actual behaviour

The browser crashes.

### Device name

ONEPLUS A6000

### Android version

OxygenOS version 11.1.2.2

### Firefox release type

Firefox Nightly

### Firefox version

106.0a1-20220831093258

### Device logs

4a9be630-bb09-44dd-808c-af4183081797
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
----
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:641)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:978)
Caused by: java.lang.reflect.InvocationTargetException
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:631)
	... 1 more
Caused by: mozilla.components.lib.state.StoreException: Exception while reducing state
	at mozilla.components.lib.state.Store$exceptionHandler$1$1.run(Store.kt:1)
	at android.os.Handler.handleCallback(Handler.java:938)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:233)
	at android.app.ActivityThread.main(ActivityThread.java:8068)
	... 3 more
Caused by: java.lang.UnsupportedOperationException: This isn't a hierarchical URI.
	at android.net.Uri.getQueryParameterNames(Uri.java:1605)
	at org.mozilla.fenix.ext.UriKt.containsQueryParameters(Uri.kt:4)
	at org.mozilla.fenix.home.blocklist.BlocklistHandler.filterContileRecentTab(BlocklistHandler.kt:8)
	at org.mozilla.fenix.home.blocklist.BlocklistMiddleware.invoke(BlocklistMiddleware.kt:20)
	at mozilla.components.lib.state.internal.ReducerChainBuilder$build$1$1.invoke(ReducerChainBuilder.kt:3)
	at mozilla.components.lib.state.internal.ReducerChainBuilder$build$threadCheck$1.invoke(ReducerChainBuilder.kt:10)
	at mozilla.components.lib.state.internal.ReducerChainBuilder$build$1$1.invoke(ReducerChainBuilder.kt:3)
	at mozilla.components.lib.state.Store$dispatch$1.invokeSuspend(Store.kt:4)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:4)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:18)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:923)

### Additional information

Notes:
- Video provided: https://gifyu.com/image/SwmMO
- Reproducible regardless of what words are typed as long as it is in the same format (eg. "hello:world" )
- Unable to access the browser until Storage & Cache is cleared

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-21444)

--- a/app/src/main/java/org/mozilla/fenix/ext/Uri.kt
+++ b/app/src/main/java/org/mozilla/fenix/ext/Uri.kt
@@ -16,7 +16,7 @@ import android.net.Uri
  * - "key=value" - Search param named "key" with value "value"
  */
 fun Uri.containsQueryParameters(searchParameters: String): Boolean {
-    if (searchParameters.isBlank()) {
+    if (searchParameters.isBlank() || this.isOpaque) {
         return false
     }
     val params = searchParameters.split("=")

--- a/app/src/test/java/org/mozilla/fenix/ext/UriTest.kt
+++ b/app/src/test/java/org/mozilla/fenix/ext/UriTest.kt
@@ -43,4 +43,14 @@ class UriTest {
         assertTrue(querySite.containsQueryParameters(searchParameters))
         assertFalse(blankQuerySite.containsQueryParameters(searchParameters))
     }
+
+    @Test
+    fun `WHEN an opaque url is checked for query parameters THEN then the result should be false`() {
+        val searchParameters = "q"
+        val opaqueUrl = Uri.parse("about:config")
+        val mailToUrl = Uri.parse("mailto:a@b.com")
+
+        assertFalse(opaqueUrl.containsQueryParameters(searchParameters))
+        assertFalse(mailToUrl.containsQueryParameters(searchParameters))
+    }
 }

